core/matrix/dense.cpp

```diff
@@ -1022,20 +1040,125 @@ std::unique_ptr<Dense<ValueType>> Dense<ValueType>::row_gather(
     return result;
 }
 
```

Comment:
And here is one empty line missing
```suggestion


```

Reply:
It is overloaded, so one empty is enough